lex.go

```diff
@@ -43,14 +43,63 @@ var tokChanCap = TokenChanCap // capacity of lexer token channel
 func SetTokenChanCap(size int) {
 	tokChanCap = size
 }
-func Lex(reader io.Reader) chan NgxToken {
+
+// LexScanner provides an interface for scanning tokens.
+type LexScanner interface {
+	// Scan is used to advance the scanner to the next token.
```

Comment:
Nit: There's an opportunity to provide some direction for the reader to help them understand how to use this type. [Stealing some bits from `bufio.Scanner.Scan`](https://pkg.go.dev/bufio#Scanner.Scan) assuming this was a concrete type and not an interface.

```suggestion
	// Scan advances the [LexScanner] to the next token, which will then be 
	// available  through the [LexScanner.Text] method. It returns false when
	// there are no more tokens, either by reaching the end of the input or an 
	// error... (there's more that could be said about it, but you get the idea)
```